packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/TransformHelper.java

```diff
@@ -60,6 +60,13 @@ public static void processTransform(ReadableArray transforms, double[] result) {
       return;
     }
 
+    float[] offsets = getTranslateForTransformOrigin(viewWidth, viewHeight, transformOrigin);
+    if (offsets != null) {
+      MatrixMathHelper.resetIdentityMatrix(helperMatrix);
+      MatrixMathHelper.applyTranslate3D(helperMatrix, offsets[0], offsets[1], offsets[2]);
+      MatrixMathHelper.multiplyInto(result, result, helperMatrix);
```

Comment:
answering https://github.com/facebook/react-native/pull/37606#discussion_r1252162579 
Multiply is required because the operation is order dependent. i.e. we need to multiply the translate operations or else the view is not translated to the transform-origin before applying other transforms 😅

We also reset the helperMatrix which behaves weird when there are multiple view transform animations running. I think it's reusing the instance between different animations.